### PR TITLE
feat(stepper): add stepper styles

### LINKED ISSUE #1338: Styles for stepper component
**Is your feature request related to a problem? Please describe.**
We need to implement the styles for the Stepper component for all Kendo themes based on the existing designs.

**Describe the solution you would like to see implemented**
Styles for the component.

**Describe the alternatives you have considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by silviyaboteva**
closes #1338.
Design: https://xd.adobe.com/view/a816c5e0-0c51-4fba-69b5-67b2dc43da5c-de27/screen/e293dfc3-6c9d-4a55-b06d-148991ddc54a/Stepper-default

### Done tasks:
- [x] Horizontal stepper
- [x] Vertical stepper
- [x] RTL support
- [x] Hovered/Focused/Disabled
- [x] Validation, e.g. invalid/success styles
- [x] Long labels support
- [x] Default Theme colors
- [x] Bootstrap Theme colors
- [x] Material Theme colors

### Things for update/fix
- [x] Vertical Stepper content styles
- [x] Design for the disabled state of all kind of steps and scenarios (passed, current, no passed yet/full approach, only labels, only indicator)
- [x] Design for the focus state of all kind of steps and scenarios (passed, current, no passed yet/full approach, only labels, only indicator)
- [x] z-index of the connector line (should be number > -1). When we have a container with a background where the stepper is placed. (Update: the connector line is removed/progressbar will be used instead).
- [x] Implement transition animation of the inline vertical content.
- [x] Implement progress bar instead on connector line and add styles for width/height.
- [x] Change classes from 
`k-step [k-step-compound k-step-first k-step-last k-step-prev k-step-next k-step-selected k-step-optional k-step-disabled k-step-error k-step-success]` to 
`k-step [k-step-first k-step-last k-step-done k-step-current k-step-optional k-step-disabled k-step-error k-step-success]` There would be not done and done steps as well as done with success and done with an error.
- [x] The optional text to be in its own span element?

### CODE REVIEW

**Comment by joneff** [comment]
On hold til we figure out prev / next vs done / not done

File: packages/default/scss/stepper/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            max-width: 10em;
+            display: inline-flex;
+            flex-wrap: wrap;
+            align-items: center;
+            justify-content: center;
+            z-index: 1;
+        }
+        .k-step-label:only-child {
+            @include border-radius( ($stepper-indicator-height + 2 * $stepper-indicator-border-width) / 2 );
+            padding: 5px 12px;
```
Comment (elena-gancheva) [inline packages/default/scss/stepper/_layout.scss:98]: variables here?

File: packages/default/scss/stepper/_layout.scss [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+        // Horizontal
+        .k-step-list-horizontal { }
+
+
+        // Vertical
+        .k-step-list-vertical {
+            .k-step-indicator + .k-step-label {
+                margin-left: 0;
+                margin-right: 12px;
```
Comment (elena-gancheva) [inline packages/default/scss/stepper/_layout.scss:219]: variable here too

File: packages/default/scss/stepper/_layout.scss [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    .k-stepper {
+        margin: $stepper-margin-y $stepper-margin-x;
+        padding: $stepper-padding-y $stepper-padding-x;
+        border-width: $stepper-border-width;
+        border-style: solid;
+        font-size: $stepper-font-size;
+        line-height: $stepper-line-height;
+        font-family: $stepper-font-family;
+        position: relative;
+        // TODO: remove those once k-widget styles are removed
```
Comment (elena-gancheva) [inline packages/default/scss/stepper/_layout.scss:13]: link the issue in the comment, related to this enhancement

File: packages/default/scss/stepper/_layout.scss [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        .k-step-label .k-step-label-optional {
+            font-style: italic;
+            flex-basis: 100%;
+        }
+
+        // Progressbar
+        .k-progressbar {
+            position: absolute;
+            pointer-events: none;
+            z-index: 0; //todo fix
```
Comment (elena-gancheva) [inline packages/default/scss/stepper/_layout.scss:125]: ?

Comment (silviyaboteva) [reply packages/default/scss/stepper/_layout.scss:125]: It was -1 before and caused an issue. So it is fixed now. I just need to remove the comment.

**Pull request merged**